include/aos/iam/nodeinfoprovider.hpp

```diff
@@ -56,6 +56,26 @@ static constexpr auto cAosComponentUM = "um";
  */
 bool IsMainNode(const NodeInfo& nodeInfo);
 
+/**
+ * Node status observer interface.
+ */
+class NodeStatusObserverItf {
+public:
+    /**
+     * Node status changed event.
+     *
+     * @param nodeID node id
+     * @param status node status
+     * @return Error
+     */
+    virtual Error NodeStatusChanged(const String& nodeID, const NodeStatus& status) = 0;
```

Comment:
I think it's better to add On prefix to observer method to state that it is a reaction on some event